components/Ink/src/private/MDCLegacyInkLayer.h

```diff
@@ -0,0 +1,102 @@
+/*
+ Copyright 2015-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <QuartzCore/QuartzCore.h>
+#import <UIKit/UIKit.h>
+
+/**
+ A Core Animation layer that draws and animates the ink effect.
+
+ Quick summary of how the ink ripple works:
+
+ 1. On touch down, blast initiates from the touch point.
+ 2. On touch down hold, it continues to spread, but will gravitate to the center point
+    of the view.
+ 3. On touch up, the ink ripple will lose energy, opacity will start to decrease.
+ */
+@interface MDCLegacyInkLayer : CALayer
+
+/** Clips the ripple to the bounds of the layer. */
+@property(nonatomic, assign, getter=isBounded) BOOL bounded;
+
+/** Maximum radius of the ink. No maximum if radius is 0 or less. This value is ignored if
+ @c bounded is set to |YES|.*/
+@property(nonatomic, assign) CGFloat maxRippleRadius;
+
+/** Set the foreground color of the ink. */
+@property(nonatomic, strong) UIColor *inkColor;
```

Comment:
Needs nullability annotations.

Reply:
Could it be defaultable?

Reply:
Done. Will not change how inkColor works in this legacy implementation since we don't want to break existing clients.